v3_MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,100 @@
+# Android Browser Switch v3 (Beta) Migration Guide
+
+See the [CHANGELOG](/CHANGELOG.md) for a complete list of changes. This migration guide outlines the basics for updating your browser switch integration from v2 to v3.
+
+_Documentation for v2 will be published to https://developers.braintreepayments.com once it is available for general release._
+
+## What's New
+
+This version of `browser-switch` simplifies the API surface and allows for more flexibility with integration patterns by giving the integrating app control over storage and retrieval of the pending browser switch request. 
+
+The minimum Android API for v3 is 23. This version of the SDK uses Java 11 and Kotlin 1.9.10.
+
+## Setup
+
+The setup of this library has not changed from v2 to v3.
+
+Add the library to your dependencies in your `build.gradle`:
+
+```groovy
+dependencies {
+  implementation 'com.braintreepayments.api:browser-switch:3.0.0-beta1'
+}
+```
+
+Then, add an `intent-filter` in the `AndroidManifest.xml` to your deep link destination activity:
+
+```xml
+<activity android:name="com.company.app.MyBrowserSwitchActivity"
+    android:exported="true">
+    ...
+    <intent-filter>
+        <action android:name="android.intent.action.VIEW"/>
+        <data android:scheme="my-custom-url-scheme"/>
+        <category android:name="android.intent.category.DEFAULT"/>
+        <category android:name="android.intent.category.BROWSABLE"/>
+    </intent-filter>
+</activity>
+```
+
+## Usage
+
+```kotlin
+class MyActivity : ComponentActivity() {
+
+    val browserSwitchClient: BrowserSwitchClient = BrowserSwitchClient()
+
+    override fun onResume() {
+        handleReturnToAppFromBrowser(intent)
+    }
+    
+    fun startBrowserSwitch() {
+        val browserSwitchOptions = BrowserSwitchOptions().apply {
+            requestCode = MY_REQUEST_CODE
+            url = "https://example.com"
+            returnUrlScheme = "my-custom-url-scheme"
+        }
+        when (val pendingRequest = browserSwitchClient.start(this, browserSwitchOptions)) {
+            is BrowserSwitchPendingRequest.Started -> { 
+                // store pending request
+            }
+            is BrowserSwitchPendingRequest.Failure -> { 
+                // browser was unable to be launched, handle failure
+            }
+        }
+    }
+    
+    fun handleReturnToAppFromBrowser(intent: Intent) {
+        // fetch stored pending request
+        storedPendingRequest()?.let { startedRequest ->
```

Comment:
```suggestion
        fetchPendingRequestFromPersistentStorage()?.let { startedRequest ->
```
🧶 : Only because `storedPendingRequest` sounds more like a property name.

Alternatively we could make `storedPendingRequest` a property in the example and have a guide in our docs on handling process kill. That way we don't have to suggest a persistent storage solution.